backend/src/core/constants.ts

```diff
@@ -43,6 +43,21 @@ export enum Ordering {
   DESC = 'DESC',
 }
 
+export enum TransactionalEmailSortField {
+  Created = 'created_at',
+  Updated = 'updated_at',
+}
+
+export interface TimestampFilter {
+  createdAt: TimestampOperator
+}
+
+export interface TimestampOperator {
+  gt?: Date
+  gte?: Date
+  lt?: Date
+  lte?: Date
+}
```

Comment:
```suggestion
export interface TimestampFilter {
  createdAt: ComparisonOperator<Date>
}

export interface ComparisonOperator<T> {
  gt?: T
  gte?: T
  lt?: T
  lte?: T
}
```

Reply:
This is not only applicable for timestamp, use generics

Reply:
good catch ser
<img width="429" alt="image" src="https://user-images.githubusercontent.com/67887489/200095455-073aa9a1-ebb0-4e92-8864-d83e6073bc05.png">